packages/react-button/src/components/Button/createButton.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { mergeProps, getSlots, simplifyShorthand } from '@fluentui/react-compose/lib/next/index';
+import { ButtonProps, ButtonState } from './Button.types';
+import { useButton } from './useButton';
+
+/**
+ * Consts listing which props are shorthand props.
+ */
+export const buttonShorthandProps = ['icon', 'loader', 'children'];
+
+/**
+ * Define the render function. Given the state of a button, renders it.
+ */
+export const renderButton = (state: ButtonState) => {
+  const { slots, slotProps } = getSlots(state, buttonShorthandProps);
+  const { loading, iconPosition, iconOnly } = state;
+
+  return (
+    <slots.root {...slotProps.root}>
+      {loading && <slots.loader {...slotProps.loader} />}
+      {iconPosition !== 'after' && <slots.icon {...slotProps.icon} />}
+      {!iconOnly && <slots.children {...slotProps.children} />}
+      {iconPosition === 'after' && <slots.icon {...slotProps.icon} />}
+    </slots.root>
+  );
+};
+
+/**
+ * Given user props, returns state and render function for a Button.
+ */
+export const createButton = (props: ButtonProps, ref: React.Ref<HTMLElement>, defaultProps?: ButtonProps) => {
+  const state = mergeProps(
+    {
+      ref,
+      as: props.href ? 'a' : 'button',
```

Comment:
I would like to keep this less magic and require from a client to pass `as='a' href='#'`

Reply:
Assuming this, it means that `<Button href="..."/>` would not work as expected by the caller. Why should the caller know that `as` should be an `a` tag to make href work correctly?

I'm trying to understand why this is better for the customer. Historically we've always had this logic even without `as`; if a user provides an href, we render the button as an `a` tag to ensure the browser behaviors work correctly.

Reply:
nevermind, after thinking about this, i am good with this

Reply:
fixed!